NitroxServer/GameLogic/Spawning/EntitySpawner.cs

```diff
@@ -1,118 +1,135 @@
+
+            foreach (PrefabData prefab in dstData.prefabs)
+            {
+                float probabilityDensity = prefab.probability / entitySpawnPoint.Density;
+                rollingProbabilityDensity += probabilityDensity;
+            }
+
+            double randomNumber = random.NextDouble();
+            double rollingProbability = 0;
+
+            if (rollingProbabilityDensity > 0)
+            {
+                if (rollingProbabilityDensity > 1f)
                 {
-                    float probabilityDensity = prefab.probability / entitySpawnPoint.Density;
-                    rollingProbabilityDensity += probabilityDensity;
+                    randomNumber *= rollingProbabilityDensity;
                 }
-                double randomNumber = random.NextDouble();
-                double rollingProbability = 0;
 
-                if (rollingProbabilityDensity > 0)
+                foreach (PrefabData prefab in dstData.prefabs)
                 {
-
-                    if (rollingProbabilityDensity > 1f)
+                    float probabilityDensity = prefab.probability / entitySpawnPoint.Density;
+                    rollingProbability += probabilityDensity;
+                    // This is pretty hacky, it rerolls until its hits a prefab of a correct type
+                    // What should happen is that we check wei first, then grab data from there
+                    bool isValidSpawn = IsValidSpawnType(prefab.classId, entitySpawnPoint.CanSpawnCreature);
+                    if (rollingProbability >= randomNumber && isValidSpawn)
                     {
-                        randomNumber *= rollingProbabilityDensity;
-                    }
-
-                    foreach (PrefabData prefab in dstData.prefabs)
-                    {
-                        float probabilityDensity = prefab.probability / entitySpawnPoint.Density;
-                        rollingProbability += probabilityDensity;
-                        //This is pretty hacky, it rerolls until its hits a prefab of a correct type
-                        //What should happen is that we check wei first, then grab data from there
-                        bool isValidSpawn = IsValidSpawnType(prefab.classId, entitySpawnPoint.CanSpawnCreature);
-                        if (rollingProbability >= randomNumber && isValidSpawn)
-                        {
-                            selectedPrefab = prefab;
-                            break;
-                        }
+                        selectedPrefab = prefab;
+                        break;
                     }
                 }
+            }
 
-                if (!ReferenceEquals(selectedPrefab, null) && worldEntitiesByClassId.ContainsKey(selectedPrefab.classId))
-                {
-                    WorldEntityInfo worldEntityInfo = worldEntitiesByClassId[selectedPrefab.classId];
+            if (!ReferenceEquals(selectedPrefab, null) && worldEntitiesByClassId.ContainsKey(selectedPrefab.classId))
```

Comment:
I know it sounds trivial, but we should stick with TryGetValue here given the number of spawn points we are iterating over. It is twice as fast as ContainsKey.